### PR TITLE
Has `mafia install` respect the MAFIA_(ALEX|HAPPY|CPPHS)=true flags.

### PR DESCRIPTION
**Pull request by damncabbage**
eg. `env MAFIA_ALEX=true MAFIA_HAPPY=true mafia install purescript`

(Paging @jystic.)

### CODE REVIEW

**Comment by damncabbage** [comment]
PS:

sorry

**Comment by thumphries** [comment]
Looks good 👲 

Wish we didn't need these env variables

**Comment by damncabbage** [comment]
Thanks. Build's still not going, despite a manual kickoff. :\

**Comment by thumphries** [comment]
ask boris?

**Comment by damncabbage** [comment]
why always boris

(Thanks; passing build, and works locally. I really don't how to write a test for this one, sorry.)

File: main/mafia.hs [hunk 1]
```diff
@@ -395,7 +395,9 @@ mafiaHoogle args = do
 
 mafiaInstall :: InstallPackage -> EitherT MafiaError IO ()
 mafiaInstall ipkg =
-  liftIO . T.putStrLn =<< firstT MafiaBinError (installBinary ipkg)
+  liftIO . T.putStrLn =<< do
+    ensureAuxillaryExes
+    firstT MafiaBinError (installBinary ipkg)
```
Comment (jacobstanley) [inline main/mafia.hs]: This is fine, but it reads a little weird, why not?

``` hs
mafiaInstall ipkg = do
  ensureAuxillaryExes
  liftIO . T.putStrLn =<< firstT MafiaBinError (installBinary ipkg)
```

**Comment by jacobstanley** [comment]
Look good to me 👍 

> Wish we didn't need these env variables

It should be possible for us to do this automatically as required, we just need to be able to read the `build-tools` section of the `.cabal` file of the package we're trying to install - maybe it's something for next yak day.

**Comment by thumphries** [comment]
> maybe it's something for next yak day

made an issue #105 , one day

**Comment by jacobstanley** [comment]
It would be great if we could do that, then we could use it for internal tools like the `sardine-compiler`, as it stands I just commit the generated file to make life easier.

File: main/mafia.hs (hunk 1 above)
Comment (damncabbage) [inline main/mafia.hs]: Rearranged; thanks.

**Pull request merged**